### PR TITLE
Test Case(s) for Component Allocations

### PR DESCRIPTION
**Pull request by kazimierzm-iRonin**

### CODE REVIEW

File: test/component_allocations_test.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	s.Equal(http.StatusCreated, subResp.Response.StatusCode)
+
+	onOffResp, err := s.client.ComponentsController().CreateComponent(
+		ctx,
+		*pf.Id,
+		models.ComponentKindPath_ONOFFCOMPONENTS,
+		interfacePtr(OneOfBody{
+			OneOfComponent: OnOffComponent{
+				Name:                      "One of component",
+				AllowFractionalQuantities: true,
```
Comment (maciej-nedza) [inline test/component_allocations_test.go:72]: could we remove `AllowFractionalQuantities` from `OneOfComponent`, and then assert for integer in allocation responses for on/off components?
`s.Equal(fmt.Sprintf("%.1f", expected[0].Quantity), *prevAllocations[0].Quantity)`

**Review by maciej-nedza — APPROVED** [review]
LGTM

File: test/component_allocations_test.go (hunk 1 above)
Comment (kazimierzm-iRonin) [reply test/component_allocations_test.go:72]: it sends back floatType but without any tenths of the number

Comment (maciej-nedza) [reply test/component_allocations_test.go]: weird, it should be int.. Thanks for verifying!

**Pull request merged**